File: InJava/app/src/main/java/com/barryzeha/quizquestion/model/di/DataModule.java
package com.barryzeha.quizquestion.model.di;

import com.barryzeha.quizquestion.model.repository.QuestionDataSource;
import com.barryzeha.quizquestion.model.repository.QuestionRepository;
import com.barryzeha.quizquestion.model.repository.QuestionRepositoryImpl;
import com.barryzeha.quizquestion.model.repository.RankingDataSource;
import com.barryzeha.quizquestion.model.repository.RankingDataSourceImpl;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;

/****
 * Project QuizQuestion
 * Created by Barry Zea H. on 01/10/2022
 * Copyright (c)  All rights reserved.
 ***/
@Module
@InstallIn(SingletonComponent.class)
public class DataModule {
    @Provides
    QuestionRepository questionRepositoryProvider(QuestionDataSource questionDataSource){
        return new QuestionRepositoryImpl(questionDataSource);
    }

}

File: InKotlin/app/src/main/java/com/barryzeha/quizquestion/di/DataModule.kt
package com.barryzeha.quizquestion.di

import com.barryzeha.quizquestion.model.repository.LocalDataSource
import com.barryzeha.quizquestion.model.repository.LocalDataSourceImpl
import com.barryzeha.quizquestion.model.repository.RankingRepository
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent

/****
 * Project QuizQuestion
 * Created by Barry Zea H. on 29/09/2022
 * Copyright (c)  All rights reserved.
 ***/
@Module
@InstallIn(SingletonComponent::class)
class DataModule {
    @Provides
    fun rankingRepositoryProvider(
        localRepository:LocalDataSource
    )=RankingRepository(localRepository)
}